src/Widgets/ListSettingsPopover.vala

```diff
@@ -1,133 +1,138 @@
-* modify it under the terms of the GNU General Public
-* License as published by the Free Software Foundation; either
-* version 3 of the License, or (at your option) any later version.
-*
-* This program is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-* General Public License for more details.
-*
-* You should have received a copy of the GNU General Public
-* License along with this program; if not, write to the
-* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
-* Boston, MA 02110-1301 USA
-*
-*/
-
-public class Tasks.ListSettingsDialog : Gtk.Dialog {
-    public E.Source source { get; construct; }
-
-    public ListSettingsDialog (E.Source source) {
-        Object (source: source);
+public class Tasks.ListSettingsPopover : Gtk.Popover {
+    private Gtk.Entry name_entry;
+
+    private E.SourceTaskList task_list;
+
+    private Gtk.RadioButton color_button_red;
+    private Gtk.RadioButton color_button_orange;
+    private Gtk.RadioButton color_button_yellow;
+    private Gtk.RadioButton color_button_green;
+    private Gtk.RadioButton color_button_blue;
+    private Gtk.RadioButton color_button_purple;
+    private Gtk.RadioButton color_button_brown;
+    private Gtk.RadioButton color_button_slate;
+    private Gtk.RadioButton color_button_none;
+
+    public E.Source _source;
+    public E.Source source {
+        set {
+            _source = value;
+            
+            name_entry.text = _source.dup_display_name ();
+            name_entry.sensitive = _source.writable;
+
+            task_list = ((E.SourceTaskList?) _source.get_extension (E.SOURCE_EXTENSION_TASK_LIST));
+            switch (task_list.dup_color ()) {
+                case "#c6262e":
+                    color_button_red.active = true;
+                    break;
+                case "#f37329":
+                    color_button_orange.active = true;
+                    break;
+                case "#e6a92a":
+                    color_button_yellow.active = true;
+                    break;
+                case "#68b723":
+                    color_button_green.active = true;
+                    break;
+                case "#3689e6":
+                    color_button_blue.active = true;
+                    break;
+                case "#a56de2":
+                    color_button_purple.active = true;
+                    break;
+                case "#8a715e":
+                    color_button_brown.active = true;
+                    break;
+                case "#667885":
+                    color_button_slate.active = true;
+                    break;
+                default:
+                    color_button_none.active = true;
+                    break;
+            }
+        }
+
+        get {
+            return _source;
+        }
+    }
+
+    public ListSettingsPopover () {
```

Comment:
Can we move these properties to the `construct` block since they're not constructor arguments?

Reply:
Of course.